src/components/modules/ChatMain/ChatMain.tsx

```diff
@@ -1,28 +1,68 @@
-import React, { useRef, useEffect } from 'react';
-
+import React, { useRef, useEffect, useContext } from 'react';
+import { SocketContext } from 'src/context/socket/context';
+import { useAppDispatch, useAppSelector } from 'src/store/hooks';
+import { selectUserProfile } from 'src/store/userSlice';
 import MessagesTable from '@modules/MessagesTable/MessagesTable';
 import ChatDayTitle from '@modules/ChatDayTitle/ChatDayTitle';
+import { ESocketEvent } from 'src/models/socket';
+import {
+  selectMessages,
+  updateListMessage,
+  selectConversation,
+} from 'src/store/conversationSlice';
 
 import './ChatMain.scss';
 
-interface IChatMain {
-  messages: string[];
-}
-const ChatMain: React.FC<IChatMain> = ({ messages }) => {
-  const scrollRef = useRef<null | HTMLDivElement>(null);
+const ChatMain: React.FC = () => {
+  const socket = useContext(SocketContext);
+  const dispatch = useAppDispatch();
+  const scrollRef = useRef<HTMLDivElement>(null);
+
+  const useProfileStore = useAppSelector(selectUserProfile);
+  const { listMessage } = useAppSelector(selectMessages);
+  const { selectedConversation } = useAppSelector(selectConversation);
   useEffect(() => {
     scrollRef.current?.scrollIntoView();
-  }, [messages]);
+  }, [listMessage]);
+  useEffect(() => {
+    socket.on(ESocketEvent.GET_MESSAGE, ({ user, message }) => {
+      if (message.conversation === selectedConversation?.id) {
```

Comment:
vì sao phải có dòng if này

Reply:
khi mình đg ở room chat này mà nhận đc tn từ room kia thì phải ràng buộc để phần socket nó k render luôn ở room đó

Reply:
phần này bên be làm rồi, room nào có tin nhắn thì nó mới nhận đượt